### PR TITLE
Canonicalize the temporary directory files

### PR DESCRIPTION
**Pull request by JaroslavTulach**
### Pull Request Description

Due to some Mac OS X symlinks trickery we need to canonicalize files created by `@Rule TemporaryFolder` to compare them for equality.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] All code follows the
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      style guides. 
- All code has been tested:
  - [x] Unit tests are now passing on Mac OS X as well

### CODE REVIEW

**Comment by JaroslavTulach** [comment]
Sample [failure](https://github.com/enso-org/enso/actions/runs/7672566748/job/20915681390?pr=8779#step:10:2832) to be fixed:
```
 [error] Test org.enso.runner.UtilsTest.detectParentProject failed: java.lang.AssertionError: Source detected expected:</var/folders/mm/pltwc2yj1jx192t6dzy9zsrr0000gn/T/junit2482077137355453939/dir/prj/src/some/file/Main.enso> but was:</private/var/folders/mm/pltwc2yj1jx192t6dzy9zsrr0000gn/T/junit2482077137355453939/dir/prj/src/some/file/Main.enso>, took 0.218 sec
  [error]     at org.enso.runner.UtilsTest.detectParentProject(UtilsTest.java:32)
  [error]     at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
  [error]     at java.lang.reflect.Method.invoke(Method.java:580)
  [error]     ...
  [error] Test org.enso.runner.UtilsTest.specifyProjectDir failed: java.lang.AssertionError: Source is the project expected:</var/folders/mm/pltwc2yj1jx192t6dzy9zsrr0000gn/T/junit6919679684730158327/dir/prj> but was:</private/var/folders/mm/pltwc2yj1jx192t6dzy9zsrr0000gn/T/junit6919679684730158327/dir/prj>, took 0.005 sec
  [error]     at org.enso.runner.UtilsTest.specifyProjectDir(UtilsTest.java:49)
  [error]     at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
  [error]     at java.lang.reflect.Method.invoke(Method.java:580)
  [error]     ...
  [error] Test org.enso.runner.UtilsTest.dontDetectParentProjectWithoutSrcDir failed: java.lang.AssertionError: Source detected expected:</var/folders/mm/pltwc2yj1jx192t6dzy9zsrr0000gn/T/junit18314718692805380716/dir/prj/nosrc/some/file/Main.enso> but was:</private/var/folders/mm/pltwc2yj1jx192t6dzy9zsrr0000gn/T/junit18314718692805380716/dir/prj/nosrc/some/file/Main.enso>, took 0.008 sec
  [error]     at org.enso.runner.UtilsTest.dontDetectParentProjectWithoutSrcDir(UtilsTest.java:100)
  [error]     at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
  [error]     at java.lang.reflect.Method.invoke(Method.java:580)
  [error]     ...
  [error] Test org.enso.runner.UtilsTest.dontDetectParentProjectIfMissingPackageYaml failed: java.lang.AssertionError: Source detected expected:</var/folders/mm/pltwc2yj1jx192t6dzy9zsrr0000gn/T/junit2179366960683506777/dir/prj/src/some/file/Main.enso> but was:</private/var/folders/mm/pltwc2yj1jx192t6dzy9zsrr0000gn/T/junit2179366960683506777/dir/prj/src/some/file/Main.enso>, took 0.009 sec
  [error]     at org.enso.runner.UtilsTest.dontDetectParentProjectIfMissingPackageYaml(UtilsTest.java:83)
  [error]     at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
  [error]     at java.lang.reflect.Method.invoke(Method.java:580)
  [error]     ...
  [error] Test org.enso.runner.UtilsTest.specifyProjectAndFile failed: java.lang.AssertionError: Source is kept expected:</var/folders/mm/pltwc2yj1jx192t6dzy9zsrr0000gn/T/junit3060233163304576085/Standalone.enso> but was:</private/var/folders/mm/pltwc2yj1jx192t6dzy9zsrr0000gn/T/junit3060233163304576085/Standalone.enso>, took 0.006 sec
  [error]     at org.enso.runner.UtilsTest.specifyProjectAndFile(UtilsTest.java:66)
  [error]     at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
  [error]     at java.lang.reflect.Method.invoke(Method.java:580)
  [error]     ...
```

**Comment by JaroslavTulach** [comment]
Mac [CI run](https://github.com/enso-org/enso/actions/runs/7676801139/job/20925722565?pr=8879) is green. Marking **ready to merge** and waiting for a code owner blessing.

**Review by GregoryTravis — APPROVED** [review]

**Review by hubertp — APPROVED** [review]

**Pull request merged**
